### PR TITLE
bugfix: Highlight for named arg in case class constructors

### PR DESCRIPTION
**Pull request by jkciesluk**
Moved from https://github.com/scalameta/metals/pull/4542

### CODE REVIEW

File: mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala [hunk 1]
```diff
@@ -131,9 +131,23 @@ final class PcDocumentHighlightProvider(
...
                   val constructorOwner =
-                    if (s.owner.owner.isCaseClass) s.owner.owner
-                    else s.owner.owner.companion
+                    if (applyOwner.isCaseClass) applyOwner
+                    else {
+                      applyOwner.companion match {
+                        case NoSymbol =>
+                          // there is a weird case, where sometimes we need to use localCompanion twice
+                          // to get correct companion class
+                          localCompanion(applyOwner) match {
```
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala:142]: Looks like first we have a module class, then module and then class.

We can do in localCompanion this instead:
```
val nameToLookFor =
        if (sym.isModuleClass) sym.name.companionName.companionName
        else sym.name.companionName
```
and use that name in lookup

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for separating this PR. I figured out what the issue is, so we don't have to lookup twice.

File: mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala (hunk 1 above)
Comment (jkciesluk) [reply mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala]: Great, thanks!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**